great_expectations/datasource/fluent/data_asset/data_connector/filesystem_data_connector.py

```diff
@@ -17,23 +17,20 @@
 
 
 class FilesystemDataConnector(FilePathDataConnector):
-    """Extension of ConfiguredAssetFilePathDataConnector used to connect to Filesystem.
-
-    Being a Configured Asset Data Connector, it requires an explicit list of each Data Asset it can
-    connect to. While this allows for fine-grained control over which Data Assets may be accessed,
-    it requires more setup.
+    """Extension of FilePathDataConnector used to connect to Filesystem (local, networked file storage (NFS), DBFS, etc.).
 
     Args:
         datasource_name: The name of the Datasource associated with this DataConnector instance
         data_asset_name: The name of the DataAsset using this DataConnector instance
         batching_regex: A regex pattern for partitioning data references
         base_directory: Relative path to subdirectory containing files of interest
         glob_directive: glob for selecting files in directory (defaults to `**/*`) or nested directories (e.g. `*/*/*.csv`)
-        data_context_root_directory: Optional GreatExpectations root directory (if installed on local filesystem)
+        data_context_root_directory: Optional GreatExpectations root directory (if installed on filesystem)
         # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
         # TODO: <Alex>ALEX</Alex>
         # sorters (list): Optional list if you want to sort the data_references
         # TODO: <Alex>ALEX</Alex>
+        file_path_template_map_fn: Format function mapping path to fully-qualified resource on filesystem (optional)
```

Comment:
Can `data_context_root_directory` & `file_path_template_map_fn` both be `None` or is at least one of them required?

Reply:
@Kilo59 This is a very good question.  The answer is that "Yes -- they can both be `None` -- though the reasons are unrelated."  Specifically, `data_context_root_directory` pertains only to the fact that `gx init` was run and the option to store the configuration on a filesystem was selected.  Specifying the root installation directory then nudges the path finder to look under there.  However, the `file_path_template_map_fn` refers to the way `Pandas` and `Spark` access files on remote systems.  In the case of `Pandas`, the files are accessed using semantic file path specification pattern (e.g., things like `/dbfs` or `"{account_name}.blob.core.windows.net/{container}/{path}"` (the latter is for Azure), while in the case of `Spark`, it is always using a URI protocol (e.g., `"wasbs://{container}@{account_name}.blob.core.windows.net/{path}"` for Azure, `"s3a://{bucket}/{path}"` for S3, and so on.  Sometimes both `Pandas` and `Spark` use the URI protocol.  For these purposes, we specify the function that formats the path that looks for the components of the path to put together, and these components are then supplied as the keyword aguements to this function by the `_get_full_file_path()` method, member of the corresponding `DataConnector` subclass.  Thank you.